File: fix_urls.js
const fs = require('fs');
const path = require('path');

const configImport = "import { API_BASE_URL } from '../config';\n";
const rootDir = path.join(__dirname, 'src');

function walkDir(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(file => {
        file = path.join(dir, file);
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) {
            results = results.concat(walkDir(file));
        } else if (file.endsWith('.jsx') || file.endsWith('.js')) {
            results.push(file);
        }
    });
    return results;
}

const files = walkDir(rootDir);

files.forEach(filepath => {
    let content = fs.readFileSync(filepath, 'utf8');
    let modified = false;

    // 1. Remove hardcoded const API_BASE_URL
    const constRegex = /const\s+API_BASE_URL\s*=\s*'https:\/\/rajugariventures\.com\/sbr-pos';\r?\n?/g;
    if (constRegex.test(content)) {
        content = content.replace(constRegex, '');
        if (!content.includes(configImport)) {
            content = configImport + content;
        }
        modified = true;
    }

    // 2. Replace hardcoded http://localhost string inside single quotes
    const localhostQuoteRegex = /'http:\/\/localhost\/pos-system([^']*)'/g;
    if (localhostQuoteRegex.test(content)) {
        content = content.replace(localhostQuoteRegex, "API_BASE_URL + '$1'");
        if (!content.includes(configImport)) {
            content = configImport + content;
        }
        modified = true;
    }

    // 3. Replace inside backticks or template literals
    if (content.includes('`http://localhost/pos-system') || (!content.includes('`${API_BASE_URL}') && content.includes('http://localhost/pos-system'))) {
        content = content.replace(/http:\/\/localhost\/pos-system/g, '${API_BASE_URL}');
        if (!content.includes(configImport)) {
            content = configImport + content;
        }
        modified = true;
    }

    const filename = path.basename(filepath);

    // 4. Leads.jsx / POS endpoints explicit paths
    if (['Leads.jsx', 'POS.jsx', 'POSLayout.jsx', 'POSDashboard.jsx', 'TaskManagement.jsx', 'Customers.jsx'].includes(filename)) {
        const routeRegex = /(axios\.(get|post|put|delete)\()(["'])(\/server\/api\/[^\3]+)\3/g;
        if (routeRegex.test(content)) {
            content = content.replace(routeRegex, "$1`${API_BASE_URL}$4`");
            if (!content.includes(configImport)) {
                content = configImport + content;
            }
            modified = true;
        }

        const urlRegex = /apiConvertUrl=(["'])(\/server\/api\/[^\1]+)\1/g;
        if (urlRegex.test(content)) {
            content = content.replace(urlRegex, "apiConvertUrl={`\\$\\{API_BASE_URL\\}$2`}"); // escape the template string in replace
            // Actually simpler:
            content = content.replace(/apiConvertUrl="\/server\/api\/convert_lead\.php"/g, "apiConvertUrl={`${API_BASE_URL}/server/api/convert_lead.php`}");
            if (!content.includes(configImport)) {
                content = configImport + content;
            }
            modified = true;
        }
    }

    if (modified) {
        fs.writeFileSync(filepath, content, 'utf8');
        console.log('Modified', filepath);
    }
});

File: fix_urls.py
import os
import re

config_import = "import { API_BASE_URL } from '../config';\n"

for root, _, files in os.walk('src'):
    for file in files:
        if file.endswith('.jsx') or file.endswith('.js'):
            filepath = os.path.join(root, file)
            with open(filepath, 'r', encoding='utf-8') as f:
                content = f.read()
            
            original_content = content
            modified = False

            # Replace const API_BASE_URL
            const_pattern = r"const\s+API_BASE_URL\s*=\s*'https://rajugariventures\.com/sbr-pos';\n?"
            if re.search(const_pattern, content):
                content = re.sub(const_pattern, "", content)
                content = config_import + content
                modified = True

            # Replace explicit http://localhost
            localhost_pattern = r"'http://localhost/pos-system([^']*)'"
            if re.search(localhost_pattern, content):
                content = re.sub(localhost_pattern, r"API_BASE_URL + '\1'", content)
                if config_import not in content:
                    content = config_import + content
                modified = True

            # Replace in backticks
            localhost_template_pattern = r"http://localhost/pos-system"
            if 'http://localhost/pos-system' in content or '${API_BASE_URL}' not in content and 'http://localhost/pos-system' in content:
                content = content.replace('http://localhost/pos-system', '')
                if config_import not in content:
                    content = config_import + content
                modified = True

            # Leads.jsx uses /server/api/leads.php without host
            if file == 'Leads.jsx':
                content = content.replace('"/server/api/leads.php"', '${API_BASE_URL}/server/api/leads.php')
                content = content.replace('"/server/api/users.php"', '${API_BASE_URL}/server/api/users.php')
                content = content.replace('"/server/api/products.php"', '${API_BASE_URL}/server/api/products.php')
                content = content.replace('/server/api/leads.php?id=', '${API_BASE_URL}/server/api/leads.php?id=')
                content = content.replace('apiConvertUrl="/server/api/convert_lead.php"', 'apiConvertUrl={${API_BASE_URL}/server/api/convert_lead.php}')
                if config_import not in content:
                    content = config_import + content
                modified = True
            
            # POS.jsx / POS dashboard endpoints often use /server/api/
            if file == 'POS.jsx' or file == 'POSLayout.jsx' or file == 'POSDashboard.jsx':
                # Just find axios calls with /server/api and prefix them if needed
                if config_import not in content:
                    content = config_import + content
                content = content.replace("axios.get('/server/api", "axios.get(${API_BASE_URL}/server/api")
                content = content.replace("axios.post('/server/api", "axios.post(${API_BASE_URL}/server/api")
                content = content.replace('axios.post("/server/api', 'axios.post(${API_BASE_URL}/server/api')
                modified = True

            if modified:
                with open(filepath, 'w', encoding='utf-8') as f:
                    f.write(content)
